rust_src/src/symbols.rs

```diff
@@ -81,3 +88,23 @@ fn fmakunbound(symbol: LispObject) -> LispObject {
     sym.set_function(LispObject::constant_nil());
     symbol
 }
+
+
+// Define this in Rust to avoid unnecessarily consing up the symbol
+// name.
+
+/// Return t if OBJECT is a keyword.
+/// This means that it is a symbol with a print name beginning with `:'
+/// interned in the initial obarray.
+#[lisp_fn]
+fn keywordp(object: LispObject) -> LispObject {
+    if object.is_symbol() {
```

Comment:
`is_symbol`/`as_symbol_or_error` checks should be replaced by `as_symbol` and pattern matching.

Reply:
Thanks, It seems `if let` is more cleaner.